repository/repo.go

```diff
@@ -29,25 +33,47 @@ type RepoCommon interface {
 
 	// GetRemotes returns the configured remotes repositories.
 	GetRemotes() (map[string]string, error)
+}
 
-	// StoreConfig store a single key/value pair in the config of the repo
+// RepoConfig is a repository able to store configurations key/value pairs.
+type RepoConfig interface {
+	// StoreConfig store a single key/value pair in the config of the repos
 	StoreConfig(key string, value string) error
```

Comment:
there is now a misleading naming here: if `StoreConfig` and friends operate on the local config only, they should be named `StoreLocalConfig`.

Reply:
Agreed since all global config functions have "Global" in their names

Reply:
I was thinking a lot about these two modes of interaction with the configuration (local/global) and thought maybe we can make them both implement the same interface.
Maybe using something like
```go
type RepoConfig interface {
	StoreConfig(key string, value string) error
	ReadConfigs(keyPrefix string) (map[string]string, error)
	ReadConfigBool(key string) (bool, error)
	ReadConfigString(key string) (string, error)
	RmConfigs(keyPrefix string) error
}

type Repo interface {
	GlobalConfig() RepoConfig
	LocalConfig() RepoConfig
	...
}
```
This can allow using easily using different storage or libraries if needed, but on the other hand it will bring more structs into the code trying to implement the `GlobalConfig()` and `LocalConfig()`, but can be easily implemented by one struct since in our case the only different between storing configuration git locally and globally is the `--global` flag 
Any thought about this design ?

Reply:
hooo, I like that !

Just name it `Config` though